### PR TITLE
Fix search by json logic not working

### PR DESCRIPTION
**Pull request by bmonjoie**
Also add support for the "in" operator for strings.

Related JIRA tickets : WC2-167

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [X] Are there enough tests

## Changes

Add import of `jsonlogic_to_q`.
Make sure that the value passed to `jsonlogic_to_q` is a JSON object.
Fix the prefix field on which the query was applied.
Add a test to ensure it is working properly.
Slight change in `jsonlogic_to_q` to include the [`in`](https://jsonlogic.com/operations.html#in) operator

## How to test

Call `/api/mobile/entitytypes/{id}/entities` with a valid `json_content` query param.

### CODE REVIEW

File: iaso/tests/fixtures/test_entity_data.xml [hunk 1]
```diff
@@ -2,11 +2,11 @@
           <start>2022-09-20T16:50:40.889+02:00</start>
           <end>2022-09-20T16:51:17.352+02:00</end>
           <register>
-            <name>Zoro Marlboro</name>
-            <father_name>Gold Marlboro</father_name>
+            <name>Don Diego de la Vega</name>
+            <father_name>Don Alejandro de la Vega</father_name>
```
Comment (quang-le) [inline iaso/tests/fixtures/test_entity_data.xml:6]: lol

File: iaso/tests/fixtures/test_entity_data2.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<data xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" id="wfp_reg_ok" version="2022081601" iasoInstance="18">
+          <start>2022-09-20T16:50:40.889+02:00</start>
+          <end>2022-09-20T16:51:17.352+02:00</end>
+          <register>
+            <name>Prince of Euphor</name>
+            <father_name>Professor Procyon</father_name>
```
Comment (quang-le) [inline iaso/tests/fixtures/test_entity_data2.xml:6]: double lol

**Review by quang-le — COMMENTED** [review]
I can't really comment on the backend code other than it seems to make sense, but I think since you enabled a new operator, the change should be reflected in the web front-end as well (either in the PR or in a separate ticket)

**Comment by bmonjoie** [comment]
@quang-le I have indeed added a new operator, but we should test it with the other kind of data (arrays, etc.), as I have only tested with text, to add it in the web front-end. This could be added in a future PR if the need for this operator in the web front-end arises.

**Review by SimonDus — APPROVED** [review]
Python wise everything make sense. Good job.  My bad on this one !

**Review by mathvdh — APPROVED** [review]
RAS c'est très bien pour moi.
Ce que je voulais dire par rapport à convertir le JSON login en Q .. 

C'est pourquoi on a pas utilisé un truc du genre [json-logic-py](https://github.com/nadirizr/json-logic-py) et puis populé les vars ... 

mais j'ai pas suivi le truc dans tous ses details donc je raconte p-e n'importe quoi

**Comment by bmonjoie** [comment]
@mathvdh Je pense que la raison pour laquelle on n'utilise pas JsonLogic-py c'est parce qu'on veut faire les requêtes en DB et que JsonLogic-py ne fait que du traitement de données en mémoire.

**Pull request merged**